### ISSUE TITLE
Framework aborts if JSON manifest parsing fails

### ISSUE THREAD

**Issue by tmwnewbold** [CppMicroServices/CppMicroServices#180]
It appears as if JSON manifests are not even syntax-checked by the resource compiler and the Framework calls abort() if parsing a manifest fails. The Framework should never, ever abort() based on external data. If parsing fails, throw an exception, refuse to install the Bundle, etc.

Try leaving a comma out of a manifest like so:

```
{
  "bundle.name" : "foo"
  "bundle.version" : "1.0.2",
  "bundle.description" : "This bundle is broken!",
  "authors" : [ "John Doe", "Douglas Reynolds", "Daniel Cannady" ],
  "rating" : 5
}

```

Add this to a bundle and attempt to install it; you'll die with something like this:

In BundlePrivate at /tmp/CppMicroServices/core/src/bundle/usBundlePrivate.cpp:66 : Parsing of manifest.json for bundle /examples/baaad failed: * Line 8, Column 3
  Missing ',' or '}' in object declaration

[  9] 0x00007fff8c60d6df abort+00000129 at /usr/lib/system/libsystem_c.dylib+386783 (no debugging symbols found)
[ 10] 0x00000001126d8ffa us::LogMsg::~LogMsg()+00000314 at /libCppMicroServices.3.0.0.dylib+94202 (no debugging symbols found)
[ 11] 0x0000000112707e38 us::BundlePrivate::BundlePrivate(std::__1::shared_ptr<us::Bundle> const&, us::CoreBundleContext*, us::BundleInfo*)+00003848 at /libCppMicroServices.3.0.0.dylib+286264 (no debugging symbols found)
[ 12] 0x00000001126fbc97 us::Bundle::Init(us::CoreBundleContext*, us::BundleInfo*)+00000103 at /libCppMicroServices.3.0.0.dylib+236695 (no debugging symbols found)
[ 13] 0x00000001127092f8 us::BundleRegistry::Register(us::BundleInfo*)+00000968 at /libCppMicroServices.3.0.0.dylib+291576 (no debugging symbols found)
[ 14] 0x00000001126ff00c us::BundleContext::InstallBundle(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)+00000140 at /libCppMicroServices.3.0.0.dylib+249868 (no debugging symbols found)

**Comment by saschazelzer** [comment]
Thanks you for the report, I agree that we need to validate the data early and not abort in case of malformed data.

**Comment by ksubramz** [comment]
Presumably, this was fixed in #119. Now, if JSON manifest parsing fails, we throw an exception

```
       try
        {
          bc.InstallBundles(testing::LIB_PATH + testing::DIR_SEP + US_LIB_PREFIX + "TestBundleKrish" + US_LIB_EXT);
        }
        catch (std::exception& e)
        {
          std::cout << "EXCEPTION: " << e.what();
        }
        std::cout << "BAR\n";
```
TestBundleKrish is a bundle with the malformed manifest.json

Output:
```
EXCEPTION: Failed to install bundle library at L:\devel\CppMicroServices_build\bin\Release\TestBundleKrish.dll: Parsing of manifest.json for bundle TestBundleKrish at L:\devel\CppMicroServices_build\bin\Release\TestBundleKrish.dll failed: * Line 3, Column 3
  Missing ',' or '}' in object declaration
BAR
```

**Comment by ksubramz** [comment]
Closing, as I can't seem to reproduce this issue
